src/Illuminate/Http/Client/PendingRequest.php

```diff
@@ -304,6 +304,32 @@ public function withHeaders(array $headers)
         });
     }
 
+    /**
+     * Add the given headers to the request if the condition evauluates to true
+     *
+     * @param  mixed  $condition
+     * @param  array  $headers
+     * @return $this
+     */
+    public function withHeadersIf(mixed $condition, array $headers)
+    {
+        $status = false;
+
+        if (is_callable($condition)) {
+            $status = (bool) $condition();
+        } else {
+            $status = (bool) $condition;
+        }
+
+        if (! $status) {
```

Comment:
Why not just `if (! (bool) value($condition)) {`

Reply:
Woah didn't know about `value()`, neat! Thanks